src/renderers/dom/shared/DOMPropertyOperations.js

```diff
@@ -158,6 +158,12 @@ var DOMPropertyOperations = {
         } else if (propertyInfo.hasBooleanValue ||
                    (propertyInfo.hasOverloadedBooleanValue && value === true)) {
           node.setAttribute(attributeName, '');
+        } else if (attributeName === 'value' && node.hasAttribute('value')) {
+          // Use loose coercion to prevent replacement on comparisons like
+          // '3e1' == 30 in Chrome (~52).
+          if (node.value != value) { // eslint-disable-line
+            node.setAttribute(attributeName, '' + value);
```

Comment:
One thing I'm not clear on yet: _Why assign the `value` attribute at all?_. 

`ReactDOMInput` will assign the `value` property in `updateWrapper`. We could do that in `mountWrapper` as well. Is it important that the attribute be assigned?